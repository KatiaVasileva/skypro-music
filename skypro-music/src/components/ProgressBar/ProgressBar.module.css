.progressBar {
    --progress-height: 5px;
    --progress-color: #ffffff;
    --progress-bg-color: #2e2e2e;

    margin: 0;
    width: 100%;
    height: var(--progress-height);
    -webkit-appearance: none;
    appearance: none;
    cursor: pointer;
    background: #2e2e2e;
    position: relative;
    overflow: hidden;
    transition: 1s;

    background: linear-gradient(to right, var(--progress-color) var(--value), var(--progress-bg-color) var(--value));
}

.progressBar:hover {
    --progress-height: 8px;
}

/* Стили для WebKit браузеров */
.progressBar::-webkit-slider-runnable-track {
    height: var(--progress-height);
    background: var(--progress-bg-color);
}

.progressBar::-webkit-slider-thumb {
    --thumb-height: 1px;
    --thumb-width: 1px;
    position: relative;
    -webkit-appearance: none;
    width: var(--thumb-width, var(--thumb-height));
    box-shadow: calc(-100vmax - var(--thumb-width, var(--thumb-height))) 0 0 100vmax var(--progress-color);
}

/* Стили для Firefox */
.progressBar::-moz-range-track {
    background: var(--progress-bg-color);
    height: var(--progress-height);
}

.progressBar::-moz-range-thumb {
    border: none;
    height: 25px;
    width: 25px;
    border-radius: 50%;
    background: var(--progress-color);
}