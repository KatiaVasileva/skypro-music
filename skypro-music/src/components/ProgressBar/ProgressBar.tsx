import { ChangeEventHandler } from "react";
import styles from "./ProgressBar.module.css";

type ProgressBarProps = {
    max: number;
    value: number;
    step: number;
    onChange: ChangeEventHandler<HTMLInputElement>;
}

function ProgressBar({ max, value, step, onChange }: ProgressBarProps) {
  return (
    <input
      className={styles.progressBar}
      type="range"
      min="0"
      max={max}
      value={value}
      step={step}
      onChange={onChange}
    />
  );
}

export default ProgressBar;
