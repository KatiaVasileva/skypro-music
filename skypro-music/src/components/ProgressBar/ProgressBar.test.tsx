import { render } from "@testing-library/react";
import ProgressBar from "./ProgressBar";
import { ChangeEvent } from "react";

describe("ProgressBar component", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <ProgressBar
        max={0}
        value={0}
        step={0}
        onChange={function (event: ChangeEvent<HTMLInputElement>): void {
          throw new Error("Function not implemented.");
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});