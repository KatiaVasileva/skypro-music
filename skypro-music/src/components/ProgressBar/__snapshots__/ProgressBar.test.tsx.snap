// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component renders correctly 1`] = `
<DocumentFragment>
  <input
    class="progressBar"
    max="0"
    min="0"
    step="0"
    type="range"
    value="0"
  />
</DocumentFragment>
`;
